### DISCUSSION TITLE
Basesections module

### DISCUSSION THREAD

**Discussion by JosePizarro3** [FAIRmat-NFDI/AreaA-data_modeling_and_schemas#157] [Interarea Topic]
Hi Area A! (@hampusnasstrom @aalbino2 @ka-sarthak @budschi )

After having played with the base sections, I have a couple of requests for `BaseSection` (just with interoperability in mind):
1. Could `lab_id` be moved to a different section specific for experiments, or be re-defined? I am not sure I fully understand the role of such id, but in simulations this will not be used. I am guessing you want to give credit to specific labs for their activities, but then it is a very specific quantity: it is the equivalent to `BaseSection` having a quantity like `program_name` defined.
2. Can `datetime` be improved? Right now it is a bit obscured what "`The date and time associated with this section`" means: is this the elapsed time, the start of the entity, or something else? In simulations we keep the elapsed time in 2 different quantities, `datetime_start` and `datetime_end`, but I am happy to hear why you decided this quantity on the first place.

It will be nice to talk about `System` as well at some point, but this we can leave it for the project meeting.

**Comment by andreaa93** [comment]
Hey, thanks for this comment. Just to explain better, lab_id is what we use to label our entries so that we can easily fish them and reference in other ones. Check out for example SystemReference class or CompositeSystemReference. 

We can discuss in a concise and focused meeting a better definition of datetime, i'm open for that! 

If you keep inspecting those classes in the future we can "interoperate" also all the system-compositesystem-puresubstance scaffolding, that i think can be made smoother

**Reply by JosePizarro3** [reply]
Yeah, still, `lab_id` is a very experiment-specific quantity to be defined in `BaseSection`, and on the normalize, it seems to me like a hacky way to resolve certain normalizations. But I must admit some of the logic is very convoluted and not so easy to understand without an example at hand :-/ maybe we can sit at some point and any of you show me some use-case.

But thanks for bringing it up, it also reminded me that we can discuss if normalization in these base classes should be reworked a bit, but we can do this in a second step: imo, `results` generation should be done in an independent module, and not in the normalize functions of classes, so we can control better what is being done and passed to `results`.

For the system, yes, I also want to understand the PureSubstance concept. This is something left on the table from when we were discussing interarea about this.

**Comment by ka-sarthak** [comment]
I was reading through base sections out of curiosity and made these observations:

-  `lab_id` is added to the `results.eln` section in the `BaseSection` class.  This allows one to use `lab_id` in the search queries to find associated entries
- `ActivityStep` (for eg., `ExperimentStep`) can make references to an `Activity` (for eg., `Experiment`) by making a query based on the `lab_id`
- `ReadableIdentifiers` section can generate `lab_id` based on entry_name, author, datetime, and mainfile name.

It seems that `lab_id` is more of a human-readable identifier (an alternative to entry_id which is a gibberish hash). Because it is human-readable, it is also human-editable, allowing an ELN user to fill in `lab_id` based on their conventions. End of the day though, the purpose is to connect sections based on it. 

I think we can use a more general name for this quantity, like `readable_identifier` for lack of a better term. By the way, I find the current description of the ~~quantity~~ `lab_id` quite helpful in understanding its role.

**Reply by JosePizarro3** [reply]
I see, so it is like a readable strings. I like the idea of keeping a readable **hash** used to identify activities, and we were already talking about it for simulations.

I like the idea of these readable hashes. I had an alternative design idea: can we redesign this as a generic (and empty of quantities) sub-section called `ReadableIdentifiers` or `ReadableHashes`, or better, `ActivityReadableIdentifiers`? I am also wondering whether this automated connection can be eased by using methods in such class.

Still, this still does not solve my initial issue which is that this is an activity-dependent section, so it shouldn't be defined in `BaseSection`.

**Reply by andreaa93** [reply]
Apart from the discussion on the exact structure of the class (we would probably fluctuate around equally good solutions), it is not an activity-based! The samples have a lab_id, and this is how their "history" card gets rendered: the activities have a sample referenced and they get knitted together thanks to that

**Reply by JosePizarro3** [reply]
Yes yes, my bad, you are putting together concepts for activities and samples in an identifier, as @ka-sarthak said.

This is kind of similar to the Encyclopedia thingy, where they defined a `material_id` to query the archive and connect entries. 

But let's chat when you are back, @aalbino2 ! 👍🏻
